### DISCUSSION TITLE
Odd behavior on Playlist sorted by popularity

### DISCUSSION THREAD

**Discussion by BuckeyePay** [Chimildic/goofy#188] [Помощь с ошибкой] [answered]
First I want to thank you for this application. It is remarkable and you have been very quick to respond and help. I am trying to spread the word.

```
function BeforeKidsMix() {
      let tracks = Source.getSavedAlbumTracks()
        
  Filter.rangeTracks(tracks, {
    album: {
      release_date: { startDate: new Date('1950.02.29'), endDate: new Date('1988.02.29')}
    }
    })
      
   Order.sort(tracks, 'meta.popularity', 'desc');       

Playlist.saveWithReplace({
      name: 'Before Kids Mix',
      tracks: tracks,
         })  
}

```
For me this playlist yields about 5,000 tracks which are initially sorted by popularity, but the popularity ticks down toward zero and  then starts again with more songs of high popularity before again descending toward zero. This reset starts randomly anywhere from about 600 to 1,000 tracks in. It then ticks down again before doing likewise several more times before the whole list is compiled.

Here's a sample of the data...

#       | song                                                                                       | artist        | popularity | random

3303 | There's A Doctor - 40th Anniversary Version - Live At   Leeds | The Who | 0 | 6003
3304 | Smash The Mirror -   40th Anniversary Version - Live At Leeds | The Who | 7 | 2477
3305 | Miracle Cure - 40th Anniversary Version - Live At Leeds | The Who | 0 | 1064
3306 | American Girl | Tom Petty and the   Heartbreakers | 71 | 3911
3307 | Got My Mind Set On You - Remastered 2004 | George Harrison | 70 | 7811
3308 | Just What I Needed | The Cars | 68 | 5211
3309 | The Way It Is | Bruce Hornsby | 68 | 1711
3310 | Need You Tonight | INXS | 64 | 3884

and then again

4341 | Street Fighting Man | The Rolling Stones | 0 | 8682
4342 | Prodigal Son | The Rolling Stones | 0 | 2125
4343 | Stray Cat Blues | The Rolling Stones | 0 | 9819
4344 | Factory Girl | The Rolling Stones | 0 | 3589
4345 | Salt Of The Earth | The Rolling Stones | 0 | 9522
4346 | The Joker | Steve Miller Band | 73 | 3354
4347 | Born to Run | Bruce Springsteen | 70 | 2555
4348 | Thunder Road | Bruce Springsteen | 63 | 6988
4349 | Head Over Heels / Broken | Tears For Fears | 56 | 9730
4350 | Badlands | Bruce Springsteen | 55 | 2346

I'm not sure if this is just the behavior because there are 5,000 tracks being sorted or if there something unusual going on.

**Comment by BuckeyePay** [comment, marked as answer]
I tried filtering the tracks for popularity ahead of the sort and that seems to take care of the problem. Thank you.

**Reply by Chimildic** [reply]
It seems like bug. What is version your instance of goofy? Try to update with last available.

**Reply by BuckeyePay** [reply]
I'm running 1.7.1 which I believe is the most current version.

**Reply by Chimildic** [reply]
I did some fixes without up version. Just copy&paste [code from library](https://script.google.com/d/1DnC4H7yjqPV2unMZ_nmB-1bDSJT9wQUJ7Wq-ijF4Nc7Fl3qnbT0FkPSr/edit?usp=sharing) to you file.
